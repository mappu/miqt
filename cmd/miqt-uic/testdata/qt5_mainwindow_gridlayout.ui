<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QPushButton" name="pushButton_2">
      <property name="text">
       <string>top left</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QPushButton" name="pushButton">
      <property name="text">
       <string>top right</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QPushButton" name="pushButton_3">
      <property name="text">
       <string>bottom left</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QPushButton" name="pushButton_4">
      <property name="text">
       <string>bottom right</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>29</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
