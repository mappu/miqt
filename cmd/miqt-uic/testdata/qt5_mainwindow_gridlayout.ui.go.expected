// Generated by miqt-uic. To update this file, edit the .ui file in
// Qt Designer, and then run 'go generate'.
//
//go:generate miqt-uic go_generate_args

package test_package_name

import (
	"github.com/mappu/miqt/qt"
)

type MainWindowUi struct {
	MainWindow    *qt.QMainWindow
	centralwidget *qt.QWidget
	gridLayout    *qt.QGridLayout
	pushButton_2  *qt.QPushButton
	pushButton    *qt.QPushButton
	pushButton_3  *qt.QPushButton
	pushButton_4  *qt.QPushButton
	menubar       *qt.QMenuBar
	statusbar     *qt.QStatusBar
}

// NewMainWindowUi creates all Qt widget classes for MainWindow.
func NewMainWindowUi() *MainWindowUi {
	ui := &MainWindowUi{}
	ui.MainWindow = qt.NewQMainWindow(nil)
	ui.MainWindow.SetObjectName("MainWindow")
	ui.MainWindow.Resize(800, 600)
	ui.centralwidget = qt.NewQWidget(ui.MainWindow.QWidget)
	ui.centralwidget.SetObjectName("centralwidget")
	ui.gridLayout = qt.NewQGridLayout(ui.centralwidget)
	ui.gridLayout.SetObjectName("gridLayout")
	ui.gridLayout.SetContentsMargins(11, 11, 11, 11)
	ui.gridLayout.SetSpacing(6)
	ui.pushButton_2 = qt.NewQPushButton(ui.centralwidget)
	ui.pushButton_2.SetObjectName("pushButton_2")

	ui.gridLayout.AddWidget2(ui.pushButton_2.QWidget, 0, 0)
	ui.pushButton = qt.NewQPushButton(ui.centralwidget)
	ui.pushButton.SetObjectName("pushButton")

	ui.gridLayout.AddWidget2(ui.pushButton.QWidget, 0, 1)
	ui.pushButton_3 = qt.NewQPushButton(ui.centralwidget)
	ui.pushButton_3.SetObjectName("pushButton_3")

	ui.gridLayout.AddWidget2(ui.pushButton_3.QWidget, 1, 0)
	ui.pushButton_4 = qt.NewQPushButton(ui.centralwidget)
	ui.pushButton_4.SetObjectName("pushButton_4")

	ui.gridLayout.AddWidget2(ui.pushButton_4.QWidget, 1, 1)
	ui.MainWindow.SetCentralWidget(ui.centralwidget) // Set central widget
	ui.menubar = qt.NewQMenuBar(ui.MainWindow.QWidget)
	ui.menubar.SetObjectName("menubar")
	ui.menubar.Resize(800, 29)
	ui.MainWindow.SetMenuBar(ui.menubar)
	ui.statusbar = qt.NewQStatusBar(ui.MainWindow.QWidget)
	ui.statusbar.SetObjectName("statusbar")
	ui.MainWindow.SetStatusBar(ui.statusbar)

	ui.Retranslate()

	return ui
}

// Retranslate reapplies all text translations.
func (ui *MainWindowUi) Retranslate() {
	ui.MainWindow.SetWindowTitle(qt.QCoreApplication_Tr("MainWindow"))
	ui.pushButton_2.SetText(qt.QWidget_Tr("top left"))
	ui.pushButton.SetText(qt.QWidget_Tr("top right"))
	ui.pushButton_3.SetText(qt.QWidget_Tr("bottom left"))
	ui.pushButton_4.SetText(qt.QWidget_Tr("bottom right"))
}
